src/client/kotlin/tech/thatgravyboat/skyblockapi/api/profile/wardrobe/WardrobeAPI.kt

```diff
@@ -0,0 +1,141 @@
+import tech.thatgravyboat.skyblockapi.helpers.McClient
+import tech.thatgravyboat.skyblockapi.impl.tagkey.ItemTagKey
+import tech.thatgravyboat.skyblockapi.modules.Module
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexGroup
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.match
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.Text.send
+import tech.thatgravyboat.skyblockapi.utils.text.TextColor
+import tech.thatgravyboat.skyblockapi.utils.text.TextProperties.stripped
+import tech.thatgravyboat.skyblockapi.utils.text.TextStyle.color
+
+@Module
+object WardrobeAPI {
+    private val wardrobeGroup = RegexGroup.INVENTORY.group("wardrobe")
+
+    private val inventoryNameRegex = wardrobeGroup.create(
+        "title",
+        "Wardrobe \\((?<currentPage>\\d+)/\\d+\\)",
+    )
+
+    private val equippedRegex = wardrobeGroup.create(
+        "equip",
+        "Slot \\d+: Equipped",
+    )
+
+
+    var inWardrobe = false
+        private set
+
+    var slots = mutableListOf<WardrobeSlot>()
+        private set
+
+    var currentSlot: Int? = null
+        private set
+
+
+    private fun processInventory(title: String, items: List<ItemStack>) {
+        var currentPage = 0
+        inventoryNameRegex.match(title, "currentPage") { (cp) ->
+            cp.toIntOrNull()?.let { currentPage = it }
+        }
+
+        for (index in 0..8) {
+            val selectStack = items[index + 36]
+            val id = 9 * currentPage + index - 8
+            var locked = false
+
+            if (selectStack.item == Items.RED_DYE) {
+                locked = true
+            } else if (equippedRegex.match(selectStack.hoverName.stripped)) {
+                currentSlot = id
+                WardrobeStorage.updateCurrentSlot(id)
+            }
+
+            val helmetStack = items[index].takeOrEmpty()
+            val chestplateStack = items[index + 9].takeOrEmpty()
+            val leggingsStack = items[index + 18].takeOrEmpty()
+            val bootsStack = items[index + 27].takeOrEmpty()
+
+            val slot = WardrobeSlot(id, mutableListOf(helmetStack, chestplateStack, leggingsStack, bootsStack), locked)
+
+            WardrobeStorage.updateSlot(slot)
+            slots = WardrobeStorage.slots
+        }
+    }
+
+    @Subscription
+    fun onInventoryUpdate(event: ContainerChangeEvent) {
+        inWardrobe = inventoryNameRegex.matches(event.title)
+
+        if (inWardrobe) processInventory(event.title, event.inventory)
+    }
+
+    @Subscription
+    fun onInventoryOpen(event: ContainerInitializedEvent) {
+        inWardrobe = inventoryNameRegex.matches(event.title)
+
+        if (inWardrobe) processInventory(event.title, event.itemStacks)
+    }
+
+    @Subscription
+    fun onInventoryClose(event: ContainerCloseEvent) {
+        inWardrobe = false
+    }
+
+    @Subscription
+    fun onProfileSwitch(event: ProfileChangeEvent) {
+        slots = WardrobeStorage.slots
```

Comment:
I dont think these should be cached here, I think it should just be a getter in a val to call the storage